### ISSUE TITLE
multiple object inference

### ISSUE THREAD

**Issue by mhyeonsoo** [IDEA-Research/Grounded-Segment-Anything#157]
Hi,
thanks for the great implementations.
I am trying to run this code with the multi-object prompt.
As you guided, I used ' . ' between each classes, but could get a bit weird results.
I have two questions with my one sample output image.

 
![image](https://user-images.githubusercontent.com/32179857/232653361-50661e61-6589-465a-bdec-5e0d3572188d.png)

1. My prompt is "asparagus . carrot . zucchini ." Is there some mistakes with this prompt? I am wondering why the model predicts all the asparagus as zucchini, and I don't think this is only because of the DINO performance issue.
2. Is there a case that we can see broken class output such as '##ni' on the image? 

Thanks again!

**Comment by tdzdog** [comment]
Same question here. I tried "cat . dog ." on asset/demo6.jpg, but I got wierd output phrase "cat dog". I also tried "cat,dog", it does return cat or dog. But the number of detected objects decreases and some objects are misclassified.

**Comment by mhyeonsoo** [comment]
Has anyone solved the issue? Not yet solved from my side yet.

**Comment by longyangqi** [comment]
> Has anyone solved the issue? Not yet solved from my side yet.

I also face this problem. 
Another question for me: I can get correct segmentation with single prompt (e.g. `cat `), however  I cannot segment out this object with multi-object prompt (e.g. like `cat . dog .`)
Have you noticed this problem?

**Comment by mhyeonsoo** [comment]
@longyangqi 
yeah, that issue is just what I am facing recently. Hope the authors are taking care of it.

**Comment by longyangqi** [comment]
@mhyeonsoo 
The "##ni" is caused by the tokenizer (something like splitting zucchini into "zucchi" + "##ni")
you can have a look at 
https://github.com/IDEA-Research/Grounded-Segment-Anything/blob/main/GroundingDINO/groundingdino/util/inference.py
